### ISSUE TITLE
How to remove filters programmatically

### ISSUE THREAD

**Issue by JasonEter** [macgile/DataGridFilter#146]
Dear macgile,

first I want to thank you for your great work on this filterable DataGrid - it's a shame that filtering and sorting is not implemented in this way in the default DataGrid control. Great job you are doing on this project!

I'm looking for a possibility to remove all filters programmatically before loading  a new ItemsSource.
The probelm is that the filters stay applyed even when I clear ItemsSource and set a new ItemsSource. My users won't understand that the filters they have set before will still be applied even after loading new items.
So I would like to remove all filters when the DataGrid is cleared (ItemsSource is set to null). 
Unfortuantely I haven't found a way to do so. I've tried searching for a "clear" or "remove" method as well as using "myDataGrid.GetFilter().Clear();". But I haven't found anything and the clear-method I tried did not work for this.
I was also inspecting the code and saw the "RemoveAllFilter" command, but this is an internal command.

Could you please tell me how correctly remove all existing filters of the FilterDataGrid contol?

Thank you very much for your support in advance.

Best regards,
Walter

**Comment by macgile** [comment]
Bonjour,
This is weird, have you tested and studied the demo app code?
For example in the general filter (above the grids) if you type an "underscore" there is no result because no last name/first name contains this character which transmits a new (empty) collection to "ItemsSource" which do not cause the problem you described

best regards

**Comment by JasonEter** [comment]
Bonjour macgile,

I had a deeper look into the code of the FilterDataGrid itself, but not into the demo app.
Today I've checked the demo app and found the difference in usage.
Instead of assigning assigning "null" to ItemsSource for clearing the DataGrid, I needed to assign a new, empty ObservableCollectiton. 

`// Instead doing this:
myDataGrid.ItemsSource = null;

// I'm now using this:
myDataGrid.ItemsSource = new ObservableCollecttion(new List<VoucherInfo>());`

Thank you very much for your response. My question (and maybe the one of some other guys out there) is answered.

Best regards
